### PR TITLE
FIX: RNN support for weight norm and `remove()`

### LINKED ISSUE #763: How to remove Weight_Normalization?
## Question

Hi !

I have successfully applied the **weight normalization wrapper** in my model, and the model can run without any bug.

However, I wanna know how to remove the weight normalization at **inference** time?

Just like the **layer normalization**，as we all know,  **layer normalization** API has a param called **is_training**, so that **layer normalization** can be removed automatically.

In a project written with pytorch called [melgan ](https://github.com/seungwonpark/melgan) 

I read their pytorch code and I found that they remove weight norm at inference time. What's more, I have pulled an [issue](https://github.com/seungwonpark/melgan/issues/37) in that project.
iuuse

## Key code in my model

```python
def wn_conv1d(x, kernel_size, channels, scope, stride=1, pad='same', dilation=1, groups=1):
  with tf.variable_scope(scope):
    output = WeightNorm(group_Conv1D(
                          filters = channels, 
                          kernel_size = kernel_size, 
                          strides = stride, 
                          padding = pad, 
                          dilation_rate = dilation,
                          groups = groups))(x)
    return output
```

## More

The **WeightNorm** is from this [repository](https://github.com/seanpmorgan/tf-weightnorm)

Haha, I found the contributor is @seanpmorgan , you sir are a hero!

In my opinion, addons should be inserted into tensorflow rather than left as a lone repository...

### PR DESCRIPTION
**Pull request by Squadrick**
* Use `recurrent_kernel` for RNNs
* Add more test cases
* Add warning for RNN + data-dependent initialization 
* Add `remove()`, fixes #763 

Continuation of #769

### CODE REVIEW

File: tensorflow_addons/layers/wrappers.py [hunk 1]
```diff
@@ -61,6 +61,11 @@ def __init__(self, layer, data_init=True, **kwargs):
         self._init_critical_section = tf.CriticalSection(name='init_mutex')
         self.is_rnn = isinstance(self.layer, tf.keras.layers.RNN)
 
+        if self.data_init and self.is_rnn:
+            print(
+                "WeightNormalization: Using `data_init=True` with RNNs is not advised"
```
[the remark this answers is not in the document]
Comment (Squadrick) [reply tensorflow_addons/layers/wrappers.py]: `logging.warn` is much nicer than `print`. Updated the PR.
